setup.py

```diff
@@ -30,6 +30,49 @@
 import unittest
 from setuptools import setup
 import setuptools.command.build_py
+import distutils.version
+import pip
+
+
+# Make sure the environment contains an up to date enough version of pip.
+PIP_VERSION = pip.__version__
+REQUIRED_PIP_VERSION = "6.0.0"
+if (
+        distutils.version.StrictVersion(PIP_VERSION)
+        <= distutils.version.StrictVersion(REQUIRED_PIP_VERSION)
+):
+    print(
+        "Your pip version is: '{}', OpenTimelineIO requires at least "
+        "version '{}'.  Please update setuptools by running: "
+        "pip install -U pip".format(
+            PIP_VERSION,
+            REQUIRED_PIP_VERSION,
+        )
+    )
```

Comment:
If it still works nicely, using `logging.error` instead of `print` or specify `file=sys.stderr` might be good for this. That way if this is running in build toolchains it will land on standard error and be more likely to escalate.